### PR TITLE
Fix ps3 build.

### PR DESCRIPTION
**Pull request by clienthax**

### CODE REVIEW

**Comment by sergiobenrocha2** [comment]
We should make it upstream friendly, i.e. libretro-common inside src/platform/libretro/

**Comment by clienthax** [comment]
@sergiobenrocha2 Moved it into there :)

**Comment by endrift** [comment]
Uh wow this is a gigantic patch for something tiny. Is all of that really needed for this? You've found a hangnail and brought out a sledgehammer.

**Comment by endrift** [comment]
If you want me to just add a byteswap function that's really not hard

**Comment by twinaphex** [comment]
He is testing this core inside the RPCS3 PlayStation emulator. The solution might not be ideal, but his main focus was to just get it up and running again. Maybe you could come up with an implementation instead? As long as @clienthax can get it to work I think he'd be fine with a less intrusive implementation.

If you can come up with a better implementation in upstream, I can hold off on merging this PR.

**Comment by endrift** [comment]
Yeah that should be fine. I have been meaning to do some fixes to get upstream building against 4.2 since I have an old PPC Mac for testing now. If I don't get to it by Sunday though I've probably lost track of it.

**Comment by twinaphex** [comment]
OK, would you like me to remind you around Monday if it has not been implemented yet or would that be too overbearing for your tastes? Let me know, I am fine with anything you suggest.

**Comment by endrift** [comment]
Oh I just meant that I won't have access to the Mac since I'll be out of town and I'd probably just want to do my replacement during the thinning out of the fork I still haven't gotten around to--so you shouldn't block merging it for any longer. Reminding me on Saturday is probably not a bad idea though.

**Comment by endrift** [comment]
Rather, blocking it past Sunday is unnecessary

**Comment by twinaphex** [comment]
OK

**Comment by clienthax** [comment]
Sounds good if it's going to be sorted upstream :)
One thing to note is that the localtime_r patch is still needed as it's not present on ps3

**Comment by clienthax** [comment]
@endrift There were a couple of issues with the merge, nothing big and looks like this can be added upstream too without any issue :)

File: src/core/log.c [hunk 1]
```diff
@@ -70,7 +70,7 @@ void mLog(int category, enum mLogLevel level, const char* format, ...) {
 	va_start(args, format);
 	if (context) {
 		if (!context->filter || mLogFilterTest(context->filter, category, level)) {
-			context->log(context, category, level, format, args);
+			(context->log)(context, category, level, format, args);
```
Comment (endrift) [inline src/core/log.c]: I don't understand why this is necessary. I do x->y(x, ...) all over the place in mGBA; why is it a problem here?

Comment (clienthax) [reply src/core/log.c]: It seems to be conflicting with something
src/core/log.c:73:55: error: macro "log" passed 5 arguments, but takes just 1

Comment (endrift) [reply src/core/log.c]: `log` isn't a macro in mGBA code. Just undef it.

Comment (clienthax) [reply src/core/log.c]: The previous defination is in the ppu maths library.. which doesn't sound like a good idea to undef.

Comment (endrift) [reply src/core/log.c]: this is a C file, not a header. it's safe to undef because nothing past the includes could possibly use it

Comment (clienthax) [reply src/core/log.c]: done

**Comment by endrift** [comment]
No complaints left, looks good 👍

**Pull request merged**